koku/koku/middleware.py

```diff
@@ -366,3 +375,32 @@ def process_request(self, request):
             request (object): The request object
         """
         setattr(request, "_dont_enforce_csrf_checks", True)
+
+
+class AccountEnhancedMetrics(Metrics):
+    def register_metric(self, metric_cls, name, documentation, labelnames=(), **kwargs):
+        if name in EXTENDED_METRICS:
+            labelnames.extend(("account",))
+        return super().register_metric(metric_cls, name, documentation, labelnames=labelnames, **kwargs)
+
+
+class AccountEnhancedMetricsBeforeMiddleware(PrometheusBeforeMiddleware):
+    metrics_cls = AccountEnhancedMetrics
+
+
+class AccountEnhancedMetricsAfterMiddleware(PrometheusAfterMiddleware):
+    metrics_cls = AccountEnhancedMetrics
+
+    def label_metric(self, metric, request, response=None, **labels):
+        new_labels = labels
```

Comment:
All of these should have docstrings for the classes and methods. The docstrings need to document how/where the classes are used. If it makes sense, they can also reference upstream docs.